cmd/kured/main.go

```diff
@@ -339,14 +343,17 @@ func drain(client *kubernetes.Clientset, node *v1.Node) {
 		IgnoreAllDaemonSets: true,
 		ErrOut:              os.Stderr,
 		Out:                 os.Stdout,
+		Timeout:             drainTimeout,
 	}
 	if err := kubectldrain.RunCordonOrUncordon(drainer, node, true); err != nil {
 		log.Fatalf("Error cordonning %s: %v", nodename, err)
 	}
 
 	if err := kubectldrain.RunNodeDrain(drainer, nodename); err != nil {
-		log.Fatalf("Error draining %s: %v", nodename, err)
+		log.Error("Error draining %s: %v", nodename, err)
+		return false
```

Comment:
There is another PR that grew a bit, and is touching this, maybe it's worth sharing efforts? 

Can you have a look at https://github.com/weaveworks/kured/pull/341/files ?

Reply:
I maybe missed something, but I think it doens't uncordon the nodes after the timeout in the #341 ?

Reply:
Correct.

Here, I think error handling can be indeed improved. We can change the function signature to return the success AND the error.

If an error happens, log it. If forceReboot isn't true, we should probably uncordon and continue the main loop. Else, the control flow continues. That makes it far clearer to read, IMO.